core/src/main/java/io/grpc/Context.java

```diff
@@ -656,7 +655,7 @@ private CancellableContext(Context parent, Deadline deadline,
         pendingDeadline = deadline.runOnExpiration(new Runnable() {
           @Override
           public void run() {
-            cancel(new TimeoutException("context timed out"));
+            cancel(Status.DEADLINE_EXCEEDED.withDescription("context timed out").asException());
```

Comment:
I made this change so we don't have to special case for `TimeoutException` in a few places, but can use `Contexts.statusFromCancelledContext` everywhere. WDYT?